### PR TITLE
[VPU] Batch support for interpolate

### PR DESCRIPTION
**Pull request by DariaMityagina**
### Details:
 - Interpolate batch support is required for the mask_rcnn_inception_v2_coco model to work

### Tickets:
 - 51846

### CODE REVIEW

File: inference-engine/tests/functional/shared_test_classes/include/shared_test_classes/single_layer/interpolate.hpp [hunk 1]
```diff
@@ -50,4 +63,13 @@ class InterpolateLayerTest : public testing::WithParamInterface<InterpolateLayer
     void SetUp() override;
 };
 
+class InterpolateLayerTestWithConfig : public testing::WithParamInterface<InterpolateLayerTestParamsWithConfig>,
```
Comment (Maxim-Doronin) [inline inference-engine/tests/functional/shared_test_classes/include/shared_test_classes/single_layer/interpolate.hpp:66]: Maybe we'll leave only one test class and change instances in all plugins? 
I don't like the current solution as far it causes a lot of ridiculous copy-pasta

Comment (DariaMityagina) [reply inference-engine/tests/functional/shared_test_classes/include/shared_test_classes/single_layer/interpolate.hpp:66]: I did it before last commit.
But I will also need to change tests in kmb plugin. Is is okay?

Comment (Maxim-Doronin) [reply inference-engine/tests/functional/shared_test_classes/include/shared_test_classes/single_layer/interpolate.hpp:66]: Why not? But you will have to use product-config repo to trigger a build with corresponding branches

File: inference-engine/src/vpu/graph_transformer/src/stages/interpolate.cpp [hunk 2]
```diff
@@ -59,8 +61,8 @@ void FrontEnd::parseInterpolate(const Model& model, const ie::CNNLayerPtr& _laye
                                  interpolateModeIt->second == InterpolateMode::LinearOnnx;
     VPU_THROW_UNLESS(modeIsSupported, "Current Interpolate supports 'nearest' and 'linear' modes only, actual {}", interpolateMode);
 
-    const auto paramIsSupported = ic == oc && in == 1 && on == 1 && isPadZeros(padsBegin) && isPadZeros(padsEnd);
-    VPU_THROW_UNLESS(paramIsSupported, "Current Interpolate does not support paddings, batches, and resize by channels");
+    const auto paramIsSupported = ic == oc && isPadZeros(padsBegin) && isPadZeros(padsEnd);
+    VPU_THROW_UNLESS(paramIsSupported, "Current Interpolate does not support paddings, and resize by channels");
```
[the remark this answers is not in the document]
Comment (DariaMityagina) [reply inference-engine/src/vpu/graph_transformer/src/stages/interpolate.cpp]: Done

File: inference-engine/tests/functional/plugin/myriad/shared_tests_instances/single_layer_tests/interpolate.cpp [hunk 3]
```diff
@@ -16,38 +16,40 @@ const std::vector<InferenceEngine::Precision> netPrecisions = {
...
 };
 
+typedef std::map<std::string, std::string> Config;
+
 const std::vector<std::vector<size_t>> inShapes = {
-        {1, 8, 38, 38},
-        {1, 8, 36, 36},
-        {1, 8, 35, 35},
-        {1, 8, 6, 6},
+        {3, 8, 38, 38},
```
[the remark this answers is not in the document]
Comment (DariaMityagina) [reply inference-engine/tests/functional/plugin/myriad/shared_tests_instances/single_layer_tests/interpolate.cpp:22]: Done

File: inference-engine/tests/functional/shared_test_classes/include/shared_test_classes/single_layer/interpolate.hpp [hunk 4]
```diff
@@ -31,16 +31,29 @@ typedef std::tuple<
...
+        InterpolateSpecificParams,
+        InferenceEngine::Precision,        // Net precision
+        InferenceEngine::Precision,        // Input precision
+        InferenceEngine::Precision,        // Output precision
+        InferenceEngine::Layout,           // Input layout
+        InferenceEngine::Layout,           // Output layout
+        InferenceEngine::SizeVector,       // Input shapes
+        InferenceEngine::SizeVector,       // Target shapes
+        LayerTestsUtils::TargetDevice,     // Device name
+        std::map<std::string, std::string> // Additional network configuration
```
[the remark this answers is not in the document]
Comment (DariaMityagina) [reply inference-engine/tests/functional/shared_test_classes/include/shared_test_classes/single_layer/interpolate.hpp:42]: Not in the common files

File: inference-engine/tests/functional/shared_test_classes/include/shared_test_classes/single_layer/interpolate.hpp (hunk 1 above)
Comment (DariaMityagina) [reply inference-engine/tests/functional/shared_test_classes/include/shared_test_classes/single_layer/interpolate.hpp]: Reverted changes with lots of copypasta

**Comment by DariaMityagina** [comment]
@Maxim-Doronin @AntonDudchenko could you please approve this request if everything is okay?

**Review by Maxim-Doronin — APPROVED** [review]

**Review by AntonDudchenko — APPROVED** [review]

**Comment by azhogov** [comment]
CI passed: https://gitlab-icv.inn.intel.com/inference-engine/product-configs/merge_requests/1688

**Pull request merged**